### ISSUE TITLE
Charts: Decide on supported web chart data formats, e.g. Plotly

### ISSUE THREAD

**Issue by jeffbl** [Shared-Reality-Lab/IMAGE-server#129]
Extracting information from graphics of charts and plots will likely always be difficult. This work item is to explore what the most prevalent/useful web standards are for charts that include a machine-readable version of the data. We've tossed around Plotly as a likely candidate, but this has not been investigated at all.

Assigning to @gp1702 since this would be a preprocessor in the vein of what is currently proposed for embedded maps, where instead of running ML on a graphic, the chart data itself would need to be put into some JSON format readable by handlers for creating renderings. Best case, perhaps this could just be standardized largely on the most common/most open format, meaning very limited translation.

**Comment by jeffbl** [comment]
@jaydeepsingh25 following up on our conversation from last week, the goal is not to ensure that we can handle a majority of charts on the entire web. It is that given a decent format, with at least a few practical examples from real sites, we can pass it along and create compelling audio/haptic renderings. The ML extraction approach in #197 may not bear fruit, so having *any* solution here will unblock the handler-side work for creating actual experiences based on complete chart information. Even if this means we only deal with a very small portion of all embedded charts on the web, it allows us to demonstrate the advantages IMAGE can bring to the table. At that point, we can also focus on evangelising the solution we pick, telling sites that if they create their charts in a certain tool, and use specific options, they will be enabling their charts to work well with IMAGE. This is part of our story for CSUN presentation. The critical thing now is to pick one where we're sure the data will be complete and accurate, create a schema (that will hopefully be usable for other underlying formats in the future), and get a prototype working so that the handler team has something to work with.

**Comment by jaydeepsingh25** [comment]
Upon initial investigation, it seems that Chartjs even more widely used than Highcharts, and may be we should support Chartjs in our extension. Comparison available at the following URLs:

- https://www.wappalyzer.com/compare/chart-js-vs-highcharts
- https://npmcharts.com/compare/highcharts,chartist,c3,chart.js,plotly,nvd3,fusioncharts,dygraphs,d3,sigma,morris.js,cytoscape,rickshaw,canvasjs,taucharts,anychart,recharts?minimal=true&interval=7

A few URLs which are using Chartjs are:

- https://tobiasahlin.com/blog/chartjs-charts-to-get-you-started/
- https://themesbrand.com/skote/layouts/charts-chartjs.html
- https://unclebigbay.w3spaces.com/data-visualization-with-chartjs-demo.html#bar-chart
- https://themes.getbootstrap.com/preview/?theme_id=19799
- https://adminlte.io/themes/v3/index3.html

**Comment by jeffbl** [comment]
Highcharts has a demonstration page here. Each one has a description on the page, and a link to the chart on the actual site. Some big names here like the NY Fed, Visa, etc. If the end sites actually allow getting the data, this is a strong plus for highcharts in my opinion:

https://www.highcharts.com/blog/posts/use-cases/

**Comment by jeffbl** [comment]
(For the chartjs links above, I find them less compelling since they are just demonstration sites that show examples of using the library, but are not anything I'd expect our user base to have any actual interest in. If we can find more compelling examples with actual data, that would help with the case for chartsjs. I'm going to go do some more poking around now...)

**Comment by jeffbl** [comment]
I've posted to the Chart.js github discussion to see if anyone has great examples:
https://github.com/chartjs/Chart.js/discussions/10126

**Comment by jeffbl** [comment]
This particular chart came up from a potential user of IMAGE. Not sure what it uses behind the scenes, or if underlying data is available. If we find an equivalent example, that may be helpful when contacting this person in the future:
https://bigcharts.marketwatch.com/quickchart/quickchart.asp?symb=SPX&insttype=&freq=1&show=True

**Comment by jeffbl** [comment]
@jaydeepsingh25 we should close this now that decision has been made, and break out any further work items as separate issues.

**Comment by jeffbl** [comment]
For the quebec.ca graphcs, which exclude Title and SubTitle information, I submitted the following via [their web form](https://www.quebec.ca/en/how-to-reach-us/general-information):

```
Here at McGill, we are creating a tool to help people who are blind or visually impaired access graphical content on the web (https://image.a11y.mcgill.ca). To showcase our work on charts and graphs, we would like to use the COVID graphs at https://www.quebec.ca/en/health/health-issues/a-z/2019-coronavirus/situation-coronavirus-in-quebec. However, in your HighCharts graphs, you do not include a Title or SubTitle, which makes it difficult to express to blind users via our tool. Can this be fixed?
```

**Comment by jeffbl** [comment]
Receveid a boilerplate response, so also sent it to info@quebec.ca, which is the mail address given for reporting accessibility issues with the quebec.ca website.

**Comment by jeffbl** [comment]
Closing in favor of #113. @jaydeepsingh25 If there are other things to be broken out from this, please make new issues, but the selection of highcharts is completed.

**Comment by jeffbl** [comment]
hamstudy.org uses highcharts for progress tracking: https://hamstudy.org/tech2018
